src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTreeNode.scala

```diff
@@ -0,0 +1,53 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.diplomaticobjectmodel.logicaltree
+
+import freechips.rocketchip.diplomacy.ResourceBindingsMap
+import freechips.rocketchip.diplomaticobjectmodel.model.OMComponent
+
+import scala.collection.mutable
+
+trait LogicalTreeNode {
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent]): Seq[OMComponent]
+}
```

Comment:
I think we should add a trait: 
```
trait HasLogicalTreeNode { this: LazyModule =>
  val logicalTree: LogicalTreeNode
  def addLogicalTreeNode(child: LogicalTreeNode): Unit = { LogicalModuleTree.add(logicalTree, child) }
}
```
that the various LazyModule classes extend. This will be the hook by which we could later automatically add edges to the logical tree graph. It will be easier if all the LogicalModules have a member with the same name, instead of `clintLogicalTree`, `rocketLogicalTree` &etc.

Reply:
ARE you suggesting that all LazyModules extend the trait HasLogicalTreeNode?